### PR TITLE
Bug 1880352: collect hostsubnet information

### PR DESCRIPTION
**Pull request by tremes**

### CODE REVIEW

**Review by natiiix — APPROVED** [review]
/lgtm

Haven't really tested it locally, but the gathering code and the test seem fine.

File: pkg/gather/clusterconfig/clusterconfig.go [hunk 1]
```diff
@@ -437,6 +446,30 @@ func GatherClusterNetwork(i *Gatherer) func() ([]record.Record, []error) {
...
 
+// GatherHostSubnet collects HostSubnet information
+//
+// The Kubernetes api https://github.com/openshift/client-go/blob/master/network/clientset/versioned/typed/network/v1/hostsubnet.go
+// Response see https://docs.openshift.com/container-platform/4.3/rest_api/index.html#hostsubnet-v1-network-openshift-io
+//
+// Location in archive: config/hostsubnet/
+func GatherHostSubnet(i *Gatherer) func() ([]record.Record, []error) {
+	return func() ([]record.Record, []error) {
+
```
Comment (natiiix) [inline pkg/gather/clusterconfig/clusterconfig.go:457]: It's a real nitpick, but is the empty line here on purpose? I.O. is generally rather condensed, so it's quite striking when there's an empty line at the beginning or the end of a code block. I don't really mind it, just pointing it out.

Comment (tremes) [reply pkg/gather/clusterconfig/clusterconfig.go:457]: I guess it's not. I'll remove. Thanks.

**Comment by martinkunc** [comment]
Thank you @tremes , it looks great.

One small detail, could you please run make gen-doc and add update of generated docs/gathered-data.md and if you have add also example of generated .json files to the archive in docs/insights-archive-sample

**Review by natiiix — APPROVED** [review]
/lgtm

**Comment by ashcrow** [comment]
Backport of https://github.com/openshift/insights-operator/pull/177

**Pull request merged**